### PR TITLE
Bug 1753124: Example Report Query YAML lack of required fields

### PR DESCRIPTION
**Pull request by dtaylor113**
Added `ReportQuery` Model & Template with required fields `spec.columns` and `spec.query`: 

image

User can now create & save Report Query example template without any errors.

### CODE REVIEW

**Comment by spadgett** [comment]
It would be good for @chancez to take a look

File: frontend/public/models/yaml-templates.ts [hunk 1]
```diff
@@ -162,8 +162,18 @@ spec:
   reportingStart: '2019-01-01T00:00:00Z'
   reportingEnd: '2019-12-30T23:59:59Z'
   runImmediately: true
-`)
-  .setIn([referenceForModel(k8sModels.DeploymentModel), 'default'], `
+`).setIn([referenceForModel(k8sModels.ReportQueryModel), 'default'], `
+apiVersion: metering.openshift.io/v1alpha1
```
Comment (chancez) [inline frontend/public/models/yaml-templates.ts]: it's `metering.openshift.io/v1`

**Comment by chancez** [comment]
I'm actually wondering if we should even have this. We have examples in our OLM CSV, and those are more complex, and require multiple other resources. I feel like this type of UI isn't very effective unless you just have a canned-list of examples you can use. Otherwise, I think a wizard style UI is more effective.

What should this example be? It's kinda unclear how we decide this. Should it be complex, with more complex query usage, or should it be trivial with very little in it?

File: frontend/public/models/yaml-templates.ts [hunk 2]
```diff
@@ -162,8 +162,18 @@ spec:
...
+kind: ReportQuery
+metadata:
+  name: example
+  namespace: openshift-metering
+spec:
+  columns:
+  - name: period_start
+    type: timestamp
+  query: SELECT * FROM
```
Comment (chancez) [inline frontend/public/models/yaml-templates.ts]: This is an invalid query. I'm a bit unclear what we're trying to achieve here.

**Comment by chancez** [comment]
If we want something trivial, i think something like this might be best:
```
apiVersion: metering.openshift.io/v1
kind: ReportQuery
metadata:
  name: example
  namespace: openshift-metering
spec:
  columns:
  - name: the_time
    type: timestamp
  query: |
    SELECT now() AS the_time;
```

If we want something less trivial, I can try to make a query which only depends on default ReportDataSources.

**Comment by dtaylor113** [comment]
Thanks @chancez, that's the example I was looking for.   Something basic for end users to see the required fields.

**Comment by spadgett** [comment]
> I feel like this type of UI isn't very effective unless you just have a canned-list of examples you can use.

We have a sidebar where we can add various examples with descriptions if you have some we can use. Try creating an Ingress to see how it looks

**Comment by chancez** [comment]
> > I feel like this type of UI isn't very effective unless you just have a canned-list of examples you can use.
> 
> We have a sidebar where we can add various examples with descriptions if you have some we can use. Try creating an Ingress to see how it looks

That sounds interesting, but I'm not seeing that when I go to the ingress page and click the create ingress button, i just see one example in the yaml editor. 

Screen Shot 2019-09-24 at 10 28 53 AM

**Comment by spadgett** [comment]
@chancez sorry, i meant network policy

Screen Shot 2019-09-24 at 1 31 56 PM

File: frontend/public/models/yaml-templates.ts [hunk 3]
```diff
@@ -162,8 +162,19 @@ spec:
   reportingStart: '2019-01-01T00:00:00Z'
   reportingEnd: '2019-12-30T23:59:59Z'
   runImmediately: true
-`)
-  .setIn([referenceForModel(k8sModels.DeploymentModel), 'default'], `
+`).setIn([referenceForModel(k8sModels.ReportQueryModel), 'default'], `
+apiVersion: metering.openshift.io/v1
```
Comment (spadgett) [inline frontend/public/models/yaml-templates.ts]: The API version here should match the API version in the model

**Comment by chancez** [comment]
@spadgett awesome, we'll have to plan to look into that for an upcoming release maybe.

File: frontend/public/models/yaml-templates.ts (hunk 3 above)
Comment (dtaylor113) [reply frontend/public/models/yaml-templates.ts]: Fixed in both places to 'v1' per @chancez comment

**Comment by dtaylor113** [comment]
@spadgett Think this is ready to go :-)  -thanks

**Pull request merged**